### DISCUSSION TITLE
Automatic backlink

### DISCUSSION THREAD

**Discussion by perevezencev** [geldata/gel#7458] [Q&A] [answered]
Hello everyone 👋 My scheme looks like this:
```
module default {
	type Category {
		required title: str;

		multi subcategories: default::Category {
			constraint exclusive;
		};

		multi products: default::Product {
			constraint exclusive;
		};

		single parent := .<subcategories[is default::Category];
	}

	type Product {
		required title: str;

		required offerId: str {
			constraint exclusive;
		};

		required single category: default::Category;
	}
}
```
when I select product categories the link is empty
[![enter image description here][1]][1]
but when I select the product category it is defined
[![enter image description here][2]][2]

Why is this happening? Should I somehow change my schema so that the links are created automatically?

  [1]: https://i.sstatic.net/A2cB7G28.png
  [2]: https://i.sstatic.net/8MIJYrGT.png

**Comment by winnerlein** [comment, marked as answer]
You need to use a computed (back)link for `products` instead of a normal link, just like you're already using a computed link for `Category.parent`.

Additionally, I recommend one-to-many links over many-to-one links, i.e. the `single link` should be the normal link, and the `many` link should be computed. In your case `Category.parent` and `Product.category` should be the normal links, and `Category.subcategories` and `Categories.products` should be computed.
